src/entry.rs

```diff
@@ -9,37 +9,55 @@ use serde_bibtex::{de::Deserializer, to_string_unchecked, validate::is_entry_key
 
 use crate::{
     db::{EntryData, RawRecordData, RecordData},
-    error::BibTeXError,
+    error::{BibtexDataError, BibtexKeyError},
 };
 
-/// A single regular entry in a BibTeX bibliography.
 #[derive(Debug, PartialEq)]
-pub struct Entry<D: EntryData> {
-    pub key: String,
-    pub record_data: D,
+pub struct BibtexKey {
+    key: String,
 }
 
-impl<D: EntryData> Entry<D> {
+impl BibtexKey {
     /// Create a new entry data with the provided key.
     ///
+    /// # Safety
+    /// The caller is required to guarantee that the key does not contain any characters which are
+    /// invalid BibTeX entry key characters, as accepted by the [`serde_bibtex::validate::is_entry_key`] method.
+    pub(crate) fn new_unchecked(key: String) -> Self {
+        Self { key }
+    }
+}
+
+impl TryFrom<String> for BibtexKey {
+    type Error = BibtexKeyError;
+
+    /// Create a new BibTeX citation key with the provided string.
     /// # Errors
     /// This method will fail if the key contains characters which are invalid BibTeX entry key
     /// characters, as accepted by the [`serde_bibtex::validate::is_entry_key`] method.
-    pub fn try_new(key: String, record_data: D) -> Result<Self, BibTeXError> {
+    fn try_from(key: String) -> Result<Self, Self::Error> {
         if is_entry_key(&key) {
-            Ok(Self::new_unchecked(key, record_data))
+            Ok(Self { key })
         } else {
-            Err(BibTeXError::InvalidKey(key))
+            Err(BibtexKeyError { key })
         }
     }
+}
 
-    /// Create a new entry data with the provided key.
-    ///
-    /// # Safety
-    /// The caller is required to guarantee that the key does not contain any characters which are
-    /// invalid BibTeX entry key characters, as accepted by the [`serde_bibtex::validate::is_entry_key`] method.
-    pub(crate) fn new_unchecked(key: String, record_data: D) -> Self {
-        Self { key, record_data }
+/// A single regular entry in a BibTeX bibliography.
+#[derive(Debug, PartialEq)]
+pub struct Entry<D: EntryData> {
+    pub key: String,
```

Comment:
Should this rather be a `BibtexKey`?

Reply:
If this is changed, I guess the `Serializer` impl will break (but is easy to fix), but hopefully nothing else.

Reply:
Also, if this is changed, for convenience, we should implement `From<BibtexKey> for String`

Reply:
If we go down this route, it should also implement `AsRef<str>` which is useful to fix the errors in `main.rs`. In which case I would even `BibtexKey` being a "newtype"-style struct.

Reply:
Oh I just realized I literally implemented this once already: https://docs.rs/serde_bibtex/latest/serde_bibtex/token/struct.EntryKey.html

But the API is quite bad, if we want to merge these changes in, then I should update upstream...

Reply:
I've added the suggested implementations. Happy to replace with `EntryKey` if you want to update upstream.